src/Std/Internal/Async/Basic.lean

```diff
@@ -13,21 +13,187 @@ namespace Internal
 namespace IO
 namespace Async
 
+/-!
+# Asynchronous Programming Primitives
+
+This module provides a set of primitives for asynchronous programming:
+
+## Monadic Interfaces
+
+- `BaseAsync`: A monadic interface for infallible asynchronous computations (alias for `EAsync Empty`)
+- `EAsync`: A monadic interface for working with asynchronous computations that may fail with error type `ε`
+- `Async`: A monadic interface for IO-based asynchronous computations that may fail with `IO.Error` (alias for `EAsync IO.Error`)
+
+## Concurrent units of work
+
+- `ExceptTask`: A task that may fail with an error of type `ε`
+- `AsyncTask`: A task that may fail with an `IO.Error` (alias for `ExceptTask IO.Error`)
+- `MaybeExceptTask`: A computation that is either immediately available or an asynchronous task
```

Comment:
Minor naming suggestion: To fit with the style established by `EIO`, `EStateM`, and `EAsync`, perhaps name `ExceptTask` -> `ETask` and `MaybeExceptTask` -> `EMaybeTask`?